lapwing-prepare-optimized-svg-for-react-component.rb

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env ruby
+
+# require 'pry'
+require 'fileutils'
+require 'nokogiri'
+
+if ARGV.size < 2
+  $stderr.puts "Usage: ruby ./lapwing-prepare-optimized-svg-for-react-component.rb source-svgs/lapwing.svg target-js/LapwingStenoDiagram.js"
+  exit 1
+end
+
+SOURCE_SVG = ARGV[0]
+TARGET_JS = ARGV[1]
+
+# #STKPWHRAO*-EUFRPBLGTSDZ
+KEYS = ["numberBarKey", "leftSKey", "leftTKey", "leftKKey", "leftPKey", "leftWKey", "leftHKey", "leftRKey", "leftAKey", "leftOKey", "starKey", "rightEKey", "rightUKey", "rightFKey", "rightRKey", "rightPKey", "rightBKey", "rightLKey", "rightGKey", "rightTKey", "rightSKey", "rightDKey", "rightZKey"]
+
+SYMBOLS = ["numberBar", "leftS", "leftT", "leftK", "leftP", "leftW", "leftH", "leftR", "leftA", "leftO", "star", "rightE", "rightU", "rightF", "rightR", "rightP", "rightB", "rightL", "rightG", "rightT", "rightS", "rightD", "rightZ"]
+
+color_config = {}
+
+KEYS.each do | key |
+  color_config["#{key}OnColor"] = "#7109AA"
+  color_config["#{key}OffColor"] = "#E9D9F2"
+end
+
+SYMBOLS.each do | key |
+  color_config["#{key}OnColor"] = "#FFFFFF"
+  color_config["#{key}OffColor"] = "#FFFFFF"
+end
+
+# This is mostly irrelevant in the front end now
+SVG_WIDTH = 140
+
+source_svg_basename = File.basename(SOURCE_SVG)
+OPTIMIZED_SVG = "./optimized-svgs/#{source_svg_basename}"
+
+if !system "node_modules/.bin/svgo --pretty --config=svgo.config.mjs -i #{SOURCE_SVG} -o #{OPTIMIZED_SVG} > /dev/null"
+  exit 1
+end
+
+@doc = File.open(OPTIMIZED_SVG) { |f| Nokogiri::XML(f) }
+
+svg = @doc.at_css "svg"
+svg["width"] = SVG_WIDTH
+
```

Comment:
If we add the following 3 lines, it will:

- A) remove the redundant `id` attribute, which is later replaced by `id={svgDiagramID}`
- B) remove `version="1.1"`
- C) remove `xml:space="preserve"`

```suggestion

svg.remove_attribute('id')
svg.remove_attribute('version')
svg.remove_attribute('space')
```

Reply:
Added with double-quotes to match surrounding code style.
![image](https://github.com/user-attachments/assets/5015d2e9-e63f-4575-b2c5-07c8e07a5ca9)